app/models/dialog_field_serializer.rb

```diff
@@ -26,6 +26,7 @@ def serialize(dialog_field, all_attributes = false)
       if category
         dialog_field.options.merge!(:category_name => category.name, :category_description => category.description)
         dialog_field.options[:force_single_value] = dialog_field.options[:force_single_value] || category.single_value
+        dialog_field.options.reject! { |key| key == :category_id } unless all_attributes
```

Comment:
@eclarizio Is this an issue for the dialog running using the api?

Reply:
Actually, I don't think it's a problem for the runner because all the runner cares about is the values, which get processed down on line 32 here. However, I do think it's a problem for the editor, because it relies on the ID of the category to determine which one is selected. The editor could probably be changed to go by name, but right now that's a bigger change than what is being proposed here.

With the other change @branic made to not fail when IDs mismatch, I think we should hold off on this PR for now.